src/routes/app-routes.tsx

```diff
@@ -1,110 +1,76 @@
+import { AllowDiagnosticsPage } from '@pages/allow-diagnostics/allow-diagnostics';
 import { BuyPage } from '@pages/buy/buy';
+import { RouteUrls } from '@routes/route-urls';
 
 export function AppRoutes(): JSX.Element {
-  const { isSignedIn: signedIn, encryptedSecretKey } = useWallet();
-  const { isOnboardingInProgress } = useOnboardingState();
-  const { search, pathname } = useLocation();
-  const setLastSeen = useUpdateLastSeenStore();
-
-  const changeScreen = useChangeScreen();
+  const { encryptedSecretKey, hasGeneratedWallet, hasSetPassword } = useWallet();
+  const { pathname } = useLocation();
   const analytics = useAnalytics();
   useSaveAuthRequest();
 
-  const isSignedIn = signedIn && !isOnboardingInProgress;
-  const isLocked = !signedIn && encryptedSecretKey;
-  const [hasAllowedDiagnostics, _] = useHasAllowedDiagnostics();
-
-  // Keep track of 'last seen' by updating it whenever a route is set.
-  useEffect(() => {
-    setLastSeen(new Date().getTime());
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [pathname]);
+  const hasOnboarded = (hasGeneratedWallet || encryptedSecretKey) && hasSetPassword;
 
   useEffect(() => {
     void analytics.page('view', `${pathname}`);
   }, [analytics, pathname]);
 
-  const getHomeComponent = useCallback(() => {
-    if (hasAllowedDiagnostics === undefined) return <AllowDiagnosticsFullPage />;
-    if (isSignedIn || encryptedSecretKey) {
-      return (
-        <AccountGate>
-          <Home />
-        </AccountGate>
-      );
-    }
-    return <Installed />;
-  }, [hasAllowedDiagnostics, isSignedIn, encryptedSecretKey]);
-
-  const getSignInComponent = () => {
-    if (isLocked) return <Unlock />;
-    if (isSignedIn)
-      return <Navigate to={RouteUrls.ChooseAccount} screenPath={RouteUrls.ChooseAccount} />;
-    return <InstalledSignIn />;
-  };
-
-  const getSignUpElement = () => {
-    if (isLocked) return <Unlock />;
-    if (isSignedIn) {
-      return (
-        <Navigate to={`${RouteUrls.ChooseAccount}${search}`} screenPath={RouteUrls.ChooseAccount} />
-      );
-    }
-    return <Installed />;
-  };
-
   return (
     <Routes>
-      <Route path={RouteUrls.Home} element={getHomeComponent()}>
-        <Route path={RouteUrls.SignOutConfirm} element={<SignOutConfirmDrawer />} />
-      </Route>
-      {/* Installation */}
-      <Route path={RouteUrls.SignInInstalled} element={<InstalledSignIn />} />
+      {/* TODO: Use a layout container route at highest level - remove PopupContainer */}
+      {/* <Route element={<Layout />}> */}
+      {hasOnboarded ? (
+        <Route
+          path={RouteUrls.Home}
+          element={
+            <AccountGate>
+              <Home />
+            </AccountGate>
+          }
+        >
+          <Route path={RouteUrls.SignOutConfirm} element={<SignOutConfirmDrawer />} />
+        </Route>
+      ) : (
+        <Route path={RouteUrls.Onboarding} element={<Onboarding />} />
+      )}
```

Comment:
Should we use conditional route declarations like this? Wouldn't it be better to conditionally navigate to these different routes?

Perhaps we need another route guard?

Reply:
I had to add this bc of how the `<VaultLoader />` works (I think). Route guards don't seem to work to keep the popup in the right state with when the wallet state is available. If I don't do this the popup gets off with the full-page. I do see this often used a pattern for the index route. I think this might have to be looked at again when we refactor the vault and how all that is working with the scripts.

Reply:
If you can think of a better idea for this, let me know. I tried just using the AccountGate but it didn't behave the same. Maybe I'll figure something out as I work on this `Layout` route (which is in-progress).

Reply:
I think I came across this. It takes a few renders before the background script's messages make it to the renderer, so it lands on the wrong route.

Maybe we can return early above all the routes, while waiting for this state? So that the route definitions themselves remain static?

Reply:
Yep, that is what I ran into, I can try that today. 👍

Reply:
Ok, I think I got this working so I'll test a bit more but then I'll push up the change to this PR so you have it.